src/containers/Header.jsx

```diff
@@ -24,8 +25,14 @@ const useStyles = makeStyles(theme => ({
   }
 }));
 
-function Header({ handleSearch }) {
+function Header({ handleSearch, setIsEditMode, setOpenAddEditModal, setSelectedMovie }) {
```

Comment:
С точки зрения хедера и кнопки, кнопке всё равное что там происходит на клик она должа запустить только колбек что получила. По этому тебе не надо имплементацию открытия модалки прокидывать сюда. Достаточно прокинуть в `handleClick` `openModalAddMovie`. [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle)

`<Header handleSearch={handleSearch} handleClick={openModalAddMovie} />`

```suggestion
function Header({ handleSearch, handleClick }) {
```